File: Algorithms/Easy/ConcatenationOfArray.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
    def getConcatenation(self, nums: list[int]) -> list[int]:
        n = len(nums)
        ans = [0] * n * 2
        for i in range(n):
            ans[i] = ans[i + n] = nums[i]

        return ans

File: Algorithms/Easy/ConcatenationOfArray.go
package easy

// Time complexity: O(n)
// Space complexity: O(1)
func getConcatenation(nums []int) []int {
	n := len(nums)
	ans := make([]int, n*2)
	for i := 0; i < n; i++ {
		ans[i], ans[i+n] = nums[i], nums[i]
	}

	return ans
}
